src/ray/core_worker/reference_count.h

```diff
@@ -324,16 +324,19 @@ class ReferenceCounter : public ReferenceCounterInterface {
   void UpdateObjectPinnedAtRaylet(const ObjectID &object_id, const ClientID &raylet_id)
       LOCKS_EXCLUDED(mutex_);
 
-  /// Check whether the object is pinned at a remote plasma store node.
+  /// Check whether the object is pinned at a remote plasma store node or
+  /// spilled to external storage. In either case, a copy of the object is
+  /// available to fetch.
   ///
   /// \param[in] object_id The object to check.
   /// \param[out] pinned_at The node ID of the raylet at which this object is
+  /// \param[out] spilled Whether this object has been spilled.
   /// pinned. Set to nil if the object is not pinned.
   /// \return True if the object exists and is owned by us, false otherwise. We
   /// return false here because a borrower should not know the pinned location
   /// for an object.
-  bool IsPlasmaObjectPinned(const ObjectID &object_id, ClientID *pinned_at) const
-      LOCKS_EXCLUDED(mutex_);
+  bool IsPlasmaObjectPinnedOrSpilled(const ObjectID &object_id, ClientID *pinned_at,
```

Comment:
Can an object be pinned and spilled at the same time? For example, if we spill but later on restore it and want to keep it pinned until another spilling decision is made

Reply:
At the moment that doesn't happen, although this interface does allow both `pinned_at` and `spilled` to return true. We can support that later, but I think it's too complicated to do right now (you'd have to choose which copy to pin if there are multiple raylets restoring).